qemu/tests/win_serial_tool_test.py

```diff
@@ -0,0 +1,77 @@
+import logging
+import re
+
+from virttest import error_context
+from virttest import utils_test
+from virttest import utils_misc
+from virttest.utils_virtio_port import VirtioPortTest
+from qemu.tests.virtio_driver_sign_check import get_driver_file_path
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    virtio serial test for windows vm:
+    1) Boot guest with virtioserialport;
+    2) Transefer data from guest to host;
+    3) Transefer data from host to guest.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    def transfer_from_host_to_guest():
+        port.open()
+        port.sock.setblocking(0)
+        port.sock.sendall(transfer_data)
+        output = session.cmd_output(guest_receive_cmd)
+        if not re.findall(guest_pattern, output, re.M):
+            test.fail("Guest fails to receive data, output is: %s" % output)
+        port.close()
+
+    def transfer_from_guest_to_host():
```

Comment:
I suggest make these two functions has a `port` param, to make it more clear.
`test`, `params` can be omitted there, because everyone develop tp-qemu should know these.
`port` is not.

Reply:
Updated, thanks